cloudinit/sources/DataSourceUpCloud.py

```diff
@@ -0,0 +1,165 @@
+# Author: Antti Myyrä <antti.myyra@upcloud.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+# UpCloud server metadata API:
+# https://developers.upcloud.com/1.3/8-servers/#metadata-service
+
+from cloudinit import log as logging
+from cloudinit import sources
+from cloudinit import util
+from cloudinit import net as cloudnet
+from cloudinit.net.dhcp import EphemeralDHCPv4, NoDHCPLeaseError
+
+
+from cloudinit.sources.helpers import upcloud as uc_helper
+
+LOG = logging.getLogger(__name__)
+
+BUILTIN_DS_CONFIG = {"metadata_url": "http://169.254.169.254/metadata/v1.json"}
+
+# Wait for a up to a minute, retrying the meta-data server
+# every 2 seconds.
+MD_RETRIES = 5
+MD_TIMEOUT = 2
```

Comment:
is the comment right?  How does 5 retries and timeout of 2 get you 60 seconds?
Maybe it is right, or maybe you used 30 before?

Reply:
Thank you for noticing this! I was testing it out with an intentionally broken metadata server previously that failed in different ways to see how it would retry, and then forgot to set it back to 30. This is now corrected.